processor/deltatocumulativeprocessor/internal/streams/mutex.go

```diff
@@ -0,0 +1,47 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package streams // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/streams"
+
+import (
+	"sync"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/identity"
+)
+
+type mtx[T any, M Map[T]] struct {
```

Comment:
IMO, locking per operation is going to add a large amount of overhead. And doing it with this wrap structure adds a decent amount of cognitive load for folks reading the code. It would be much simpler to take a lock at a higher level. IE at `ConsumeMetrics()` or `Aggregate()`